pandas/io/pytables.py

```diff
@@ -4289,9 +4275,8 @@ def write_data(self, chunksize: Optional[int], dropna: bool = False):
 
         # consolidate masks
         if len(masks):
-            mask = masks[0]
-            for m in masks[1:]:
-                mask = mask & m
+            combine_masks = lambda first, second: first & second
+            mask = reduce(combine_masks, masks)
```

Comment:
not a deal-breaker, but i dont find this clearer than what we have now

Reply:
OK, I supposed it may be a questionable change. I will revert this.